crypto/ec/ec_lcl.h

```diff
@@ -120,6 +120,23 @@ struct ec_method_st {
      * EC_POINT_have_precompute_mult (default implementations are used if the
      * 'mul' pointer is 0):
      */
+    /*-
+     * mul() calculates the value
+     *
+     *   r := generator * scalar
+     *        + points[0] * scalars[0]
+     *        + ...
+     *        + points[num-1] * scalars[num-1].
+     *
+     * For a fixed point multiplication (scalar != NULL, num == 0)
+     * or a variable point multiplication (scalar == NULL, num == 1),
+     * mul() must use a constant time algorithm: in both cases callers
+     * should provide an input scalar (either scalar or scalars[0])
+     * in the range [0, ec_group_order); for robustness, implementers
+     * should handle the case when the scalar has not been reduced, but
+     * may treat it as an unusual input, without any constant-timeness
+     * guarantee.
```

Comment:
@mattcaswell 

> But I'm thinking from a caller perspective not much has changed, except to say the padding of the scalar input for `EC_POINT_mul` is no longer required - but we never said it was in the first place (and if anyone is doing it now and continues to do it afterwards then, that's unnecessary but - no big deal). Any public API change we make must be 100% backwards compatible.

If callers keep doing the padding it is not only unnecessary, but constant-timeness is not guaranteed, which is a big deal once we delegate constant-timeness  for fixed and variable point multiplication to `EC_POINT_mul`.

This is actually the change that we wanted to make explicit!

It is 100% backwards binary compatible with existing alternative `EC_METHODs` (cf. https://github.com/openssl/openssl/pull/6070#issuecomment-384311256 and https://github.com/openssl/openssl/pull/6070#issuecomment-384248024), but needs to be stated for future `EC_METHOD` implementations.